Providers/Scripts/2.4x-2.5x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,295 @@
+
+class TestOMSAgent(IOMSAgent):
+    def restart_oms_agent(self):
+        return True
+
+OMS_ACTION = OMSAgentUtil()
+
+def Set_Marshall(WorkspaceID,Ensure):
+    global CONF_PATH
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
+    CONF_PATH = CONF_PATH.replace('<WorkspaceID>',WorkspaceID)
+    Ensure = Ensure.encode('ascii', 'ignore')
+    return Set(WorkspaceID, Ensure)
+
+def Test_Marshall(WorkspaceID,Ensure):
+    global CONF_PATH
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
+    Ensure = Ensure.encode('ascii', 'ignore')
+    CONF_PATH = CONF_PATH.replace('<WorkspaceID>',WorkspaceID)
+    return Test(WorkspaceID, Ensure)
+
+def Get_Marshall(WorkspaceID,Ensure):
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
+    Ensure = Ensure.encode('ascii', 'ignore')
+
+    retval = 0
+    retd = dict()
+    retd['WorkspaceID'] = WorkspaceID
+    retd['Ensure'] = Ensure
+    return retval, retd
+ 
+def Set(WorkspaceID,Ensure):
+    plugin_dir = os.path.join(MODULE_RESOURCE_DIR, "plugin")
+    conf_dir = os.path.join(MODULE_RESOURCE_DIR, "conf")
+    lib_dir = os.path.join(MODULE_RESOURCE_DIR, "lib")
+    reg_dir = os.path.join(MODULE_RESOURCE_DIR, "reg")
+    yaml_dir = os.path.join(MODULE_RESOURCE_DIR, "yaml")
+
+    if Ensure == 'Present':
+        # copy all the required files to the client machine
+        copy_all_files(conf_dir, CONF_PATH)
+        update_state_dir_path(os.path.join(CONF_PATH, "container.conf"), WorkspaceID)
+        copy_all_files(plugin_dir, PLUGIN_PATH)
+        copy_all_files(lib_dir, LIB_PATH)
+        copy_all_files(reg_dir, REG_PATH)
+        check_and_create_yaml(yaml_dir, YAML_PATH)
+    elif Ensure == 'Absent':
+        # and delete CONF file in the directory
+        delete_all_files(conf_dir, CONF_PATH)
+    else:
+        # log error Ensure value not expected
+        LG().Log('ERROR', "Ensure value: " + Ensure + " not expected")
+        return [-1]
+    
+    # restart oms agent
+    if OMS_ACTION.restart_oms_agent(WorkspaceID):
+        return [0]
+    else:
+        return [-1]
+
+def Test(WorkspaceID, Ensure):
+    """
+    Test method for the DSC resoruce
+    If it returns [0] no further action is taken
+    If it returns [-1] Set_Marshall is called
+    """
+
+    # test for the existence of plugin and conf subfolders in the current plugin
+    plugin_dir = os.path.join(MODULE_RESOURCE_DIR, "plugin")
+    conf_dir = os.path.join(MODULE_RESOURCE_DIR, "conf")
+    lib_dir = os.path.join(MODULE_RESOURCE_DIR, "lib")
+    reg_dir = os.path.join(MODULE_RESOURCE_DIR, "reg")
+    yaml_dir = os.path.join(MODULE_RESOURCE_DIR, "yaml")
+    
+    if Ensure == 'Present':
+        # check all files exist
+        if (not check_all_files(plugin_dir, PLUGIN_PATH) or not check_all_files(conf_dir, CONF_PATH) or not check_all_files(lib_dir, LIB_PATH) or not check_all_files(reg_dir, REG_PATH) or not check_all_files(yaml_dir, YAML_PATH)):
+            return [-1]
+    elif Ensure == 'Absent':
```

Comment:
In this case, shouldn't you check that the relevant configuration files are not present? In the Set_Marshall case, you remove them, so you should check here if that needs to be called.